### ISSUE TITLE
Port to the Sandstorm platform

### ISSUE THREAD

**Issue by GitHubGeek** [wallabag/wallabag#1160]
[Sandstorm](https://sandstorm.io/), the personal web app platform, seems to be a perfect match.

A pleasant side-effect is that https://github.com/wallabag/wallabag/issues/769 would be a non-issue because Sandstorm handles uesr auth. [Sandstorm porting guide](https://github.com/sandstorm-io/sandstorm/wiki/Porting-Guide).

**Comment by Miladiir** [comment]
Isn't this completely this completely unnecessary overhead for an application running with LAMP?

**Comment by rpdillon** [comment]
Sandstorm provides an easy mechanism to install Wallabag (one-click), as well and providing a strong layer of security between hosted apps and the underlying platform that uses a single sign-on mechanism.

**Comment by Miladiir** [comment]
Why would I want single sign on with a self hosted application? What I want f of Walla bag, and what I got up to this point, was Get pocket Pro, on my server, for free. I'd appreciate it if you guys don't overdo it on features. All Walla bag really needs are some fixes for specific websites, syncing unread state on mobile applications and that's it.






On Tue, Jun 2, 2015 at 9:09 AM -0700, "Rick Dillon" <notifications@github.com> wrote:












Sandstorm provides an easy mechanism to install Wallabag (one-click), as well and providing a strong layer of security between hosted apps and the underlying platform that uses a single sign-on mechanism.

**Comment by tcitworld** [comment]
wallabag is a project where everyone is welcomed to participate, so if anyone wants to develop this feature, it will be welcomed, even if it's one of us.

**Comment by ocdtrekkie** [comment]
Heh, found this via HN when I suggested I'd love to use this on my Sandstorm. @paulproteus would love to see it.

Packaging an app for Sandstorm is super simple, and it'd probably be pretty easy for someone to do it, though generally it's 'preferred' if the app developers maintain an app directly. There's been a lot of work on ensuring a Sandstorm package can be built without maintaining a separate branch of code or anything as well.

Regardless of who does it, I hope someday to see it, because it looks like a pretty awesome application!

**Comment by nicosomb** [comment]
Will have a look on sandstorm documentation.

**Comment by paulproteus** [comment]
Hi @nicofrand !

FWIW, I co-wrote the Sandstorm packaging docs and Sandstorm packaging tooling. You can read the PHP-oriented tutorial here:

https://docs.sandstorm.io/en/latest/vagrant-spk/packaging-tutorial/

and I'm very happy to answer any questions.

**Comment by jibe-b** [comment]
As a user of sandstorm, I'd love to switch from my remote hosted account to one in my self-hosted sandstorm!

Thanks a lot for considering this option!

**Comment by QJKX** [comment]
It would help if #2800 used webkeys for API access (https://docs.sandstorm.io/en/latest/developing/http-apis/#webkeys). That way all clients would *just work* with a sandstorm port, and the sandstorm port would have the security benefit of the endpoint being gatekeeped by sandstorm instead of wallabag.

**Comment by ocdtrekkie** [comment]
@QJKX Indeed. It can be very frustrating if we get a Sandstorm app package working, and the client apps don't generally support working with it. The link mentions both methods that commonly work well with Sandstorm. I know with TTRSS, most clients that support HTTP Basic Auth (something TTRSS can be configured to use natively) work well with the Sandstorm package.

As a note, Sandstorm development has picked up again, and I've been working heavily on vagrant-spk, Sandstorm's app packaging tool. If anyone is interested in working on this package and needs any help or guidance, feel free to hit me up.
